psydac/linalg/tests/test_linalg.py

```diff
@@ -429,7 +454,7 @@ def test_square_block_basic(n1, n2, p1, p2, P1=False, P2=False):
     assert B**1 == B
```

Comment:
I can change that. What exactly is the difference between these two? Does `is` always check whether two objects are the same while `==` can be overwritten?

Reply:
Writing `a is b` is equivalent to `id(a) == id(b)`. The binary operator `is` checks that the two variables point to the same object in memory, and it cannot be overwritten.

Equality (`==` or `__eq__`) checks for some kind of mathematical equality and it can be overwritten. `a is b` implies `a == b`, but not vice versa.